images/virtualization-artifact/pkg/monitoring/metrics/virtualmachine/collector.go

```diff
@@ -17,40 +17,135 @@ limitations under the License.
 		"The virtualmachine current phase.",
-		[]string{"name", "namespace", "uid", "node", "phase"},
-		nil),
+		WithBaseLabels("phase"),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationCpuCores: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationCpuCores),
+		"The virtualmachine current core count.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationCpuCoreFraction: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationCpuCoreFraction),
+		"The virtualmachine current coreFraction.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationCpuRequestedCores: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationCpuRequestedCores),
+		"The virtualmachine current requested cores.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationCpuRuntimeOverhead: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationCpuRuntimeOverhead),
+		"The virtualmachine current cpu runtime overhead.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationMemorySize: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationMemorySize),
+		"The virtualmachine current memory size.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationMemoryRuntimeOverhead: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationMemoryRuntimeOverhead),
+		"The virtualmachine current memory runtime overhead.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineAwaitingRestartToApplyConfiguration: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineAwaitingRestartToApplyConfiguration),
+		"The virtualmachine awaiting restart to apply configuration.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationApplied: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationApplied),
+		"The virtualmachine configuration applied.",
+		WithBaseLabels(),
+		nil,
+	),
+
+	MetricVirtualMachineConfigurationRunPolicy: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachineConfigurationRunPolicy),
+		"The virtualmachine current runPolicy.",
+		WithBaseLabels("runPolicy"),
+		nil,
+	),
+	MetricVirtualMachinePod: prometheus.NewDesc(prometheus.BuildFQName(metrics.MetricNamespace, "", MetricVirtualMachinePod),
+		"The virtualmachine current active pod.",
+		WithBaseLabels("pod"),
+		nil,
+	),
 }
 
-func SetupCollector(vmLister Lister, registerer prometheus.Registerer) *Collector {
+func SetupCollector(reader client.Reader, registerer prometheus.Registerer) *Collector {
```

Comment:
Consider split this file into 3: collector.go with Collector, metrics.go with metrics definitions and a scraper.go with the scraper.